### ISSUE TITLE
Vertically center elements with XY Grid

### ISSUE THREAD

**Issue by Snorvarg** [zurb/foundation-sites#10388]
Hi guys, 
I try to master the XY Grid stuff, and want to y-center the content in several horizontally aligned cells, but I can't seem to fix it! 

I have a table-like structure, with several rows. Every rows looks like this piece of html: 

```
<div class="grid-x">
  <div class="small-1 cell">content that need to be vertically centered!</div>
  <div class="small-1 cell">Same here</div>
  <div class="small-1 cell">image that need to be centered vertically as well</div>
</div>
```

It looks real nice, but I can't find out how to center their content vertically.

It seem easy to do with the flex-grid: 
http://foundation.zurb.com/sites/docs/flex-grid.html#vertical-alignment

...But can I mix a flex-grid into an XY grid?!

Thank's for any help!

**Comment by jashwant** [comment]
`x-y grid` uses `display:flex` property in it's core. So, you can use anything which relies on `flex` layout. So, yes you can use alignment classes mentioned in flex-grid section. 

doc is somewhat misleading but if you will look at the start of [x-y grid](http://foundation.zurb.com/sites/docs/xy-grid.html ) section

> The XY grid works very similarly to the standard float grid, but includes a number of useful features only possible with Flexbox, like horizontal and vertical alignment, automatic sizing and a full vertical grid.

**Comment by brettsmason** [comment]
@Snorvarg As @jashwant said, all flexbox based alignments work the same as they did with the legacy flexbox grid. So you just need to add `align-middle` to the `grid-x` element.

**Comment by Snorvarg** [comment]
It was too simple to even try it! 

This works, almost: 
```
<div class="grid-x align-middle">
  <div class="small-1 cell">content that need to be vertically centered!</div>
  <div class="small-1 cell">Same here</div>
  <div class="small-1 cell">image that need to be centered vertically as well</div>
</div>
```

They get centered alright, but it is the cell-block which gets centered. I would like the content of the cell-block to get centered. I think the image below explains it pretty well:

[URL=http://www.image-share.com/ipng-3574-93.html][IMG]http://www.image-share.com/upload/3574/93.png[/IMG][/URL]

You think there is an easy fix to that?

----

You mean the packages have similar functionality, maybe you could organize the docs so "shared/equal" functionality is shown somehow, and the specifics for each package is shown?

**Comment by Snorvarg** [comment]
@brettsmason, I saw your link to the flexbox utilities now, thank you, I start to get how the docs are working now. :)

**Comment by Snorvarg** [comment]
Sorry, I should read before I write! :) The align-stretch was doing exactly what I wanted:

```
<div class="grid-x align-stretch">
  <div class="small-1 cell">content that need to be vertically centered!</div>
  <div class="small-1 cell">Same here</div>
  <div class="small-1 cell">image that need to be centered vertically as well</div>
</div>
```

Thanks for getting me started!

**Comment by Snorvarg** [comment]
The align-stretch would stretch each cell to equal size, but the inner content is not centered. (Thought I fixed it, but no) Any easy way to tell the content to stay centered?

**Comment by brettsmason** [comment]
@Snorvarg Could you add a CodePen with what you are trying to do? Hard to understand fully from a photo.

**Comment by Snorvarg** [comment]
I could work around it with a fixed padding-top of 35px in the text-blocks. The image to the left defines the height of the row anyway. 

See this pic with the Inspector showing the html and css: 
http://www.image-share.com/upload/3575/39.png
Easier than extracting the vital parts into CodePen, hope it's ok. :)

Still, it could be nice to have a 'zurb-way' of solving it.

**Comment by brettsmason** [comment]
What you are trying to create is a table - not sure why you can't use a table for this?

Anyway all you need to do is set the `cell` to a flex parent too, you can either use the Foundation class `.flex-container`, and then `align-middle` on each cell, or I would personally just write your own CSS.
From what I can see in your screenshot, something like:

```
.score-row .cell {
  display: flex;
  align-items: center;
}
```

There doesn't need to be a Zurb way to solve everything 😄

**Comment by Snorvarg** [comment]
Brettsmason, that is great! I will try your idea to get rid of my ugly padding-top. 

There is a reason I am not using a table. I have another similar view where I am using table, and it is looking super, except the elements inside a table-row affect each other! 

See this image: 
http://www.image-share.com/upload/3575/82.png

The div-elements forming the multi-colored donut push down the content of the other columns! Reason? Pieces of donut are floating. At least what I think. (There was more to it, but I have already forced myself to forget it)

So I decided to try with a table-less solution, and so far it has been working better.
